Revit_Core_Engine/Convert/Environment/FromRevit/CableTray.cs

```diff
@@ -79,9 +89,58 @@ public static List<BH.oM.MEP.System.CableTray> CableTrayFromRevit(this Autodesk.
                     StartPoint = segment.StartPoint(),
                     EndPoint = segment.EndPoint(),
                     SectionProperty = sectionProperty,
-                    ConnectionProperty = null,
+                    ConnectionProperty = new CableTrayConnectionProperty(),
                     OrientationAngle = orientationAngle
                 };
+                Vector bhomCableTrayVector = BH.Engine.Geometry.Modify.RoundCoordinates((thisSegment.StartPoint - thisSegment.EndPoint),4).Normalise();
+
+                if (queried.Count > 1)
+                {
+                    if (i == 0)
+                    {
+                        if (revitCableTrayVector == bhomCableTrayVector)
```

Comment:
This equality query is dangerous. I would rather use `BH.Engine.Geometry.Query.IsEqual` with tolerance.